{"cells":[{"cell_type":"markdown","metadata":{"id":"XUmYvAk6yri8"},"source":["# UMR MBT Microbiome Praktikum\n","\n","Scripts and tutorials for analyzing microbiome data. Lab practice for the lecture: [Moderne molekulare und Hochdurchsatz-Technologien in der medizinischen Grundlagenforschung](https://lsf.uni-rostock.de/qisserver/rds?state=verpublish&status=init&vmfile=no&publishid=144145&moduleCall=webInfo&publishConfFile=webInfo&publishSubDir=veranstaltung)\n","\n","## Description\n","\n","These sessions will cover the use of a variety of software tools needed for the analysis of microbiome data, from the handling of the Illumina sequencing data, to the processing of 16S rRNA amplicon data. During these sessions, the students will be able to:\n","\n","* Evaluate the quality of an Illumina sequencing run, including data filtering;\n","* Carry out assemblies of 16S rRNA amplicons;\n","* Assign Illumina runs to OTUs from 16S rRNA databases;\n","* Learn the basics on the R programming language and environment; and\n","* State how to manipulate microbiome data including count tables and sample metadata.\n","\n","\n","##### Venues and Dates\n","\n","\n","* Wednesday October 18th and 25th 2023 (10:00 - 11:30), at the SR2 - ZIM (Zentrum für Innere Medizin, Ernst-Heydemann-Str. 6)\n","\n","#### Presentations\n","\n","* Slides: [MBTPraktikum2023V01.pdf](https://drive.google.com/file/d/1IvdyRI0kiJNK5ECtGWHwOgrY763N10QR/view?usp=share_link)\n","\n","#### Software\n","\n","* All required software, packages and data are accessible through our virtual Binder environment\n","\n","---\n","\n","## Session 1\n","\n","\n","##### 1.1 Setup and introduction to the command line for Bioinformatics\n","\n","First, let's run the following cell, to install the `conda` package manager\n"]},{"cell_type":"code","source":["! wget https://raw.githubusercontent.com/barrantesisrael/mbtmicrobiome2023/main/materials/setup_conda\n","%run setup_conda"],"metadata":{"id":"bgY2B4gZOnfS","executionInfo":{"status":"ok","timestamp":1698161921831,"user_tz":-120,"elapsed":20556,"user":{"displayName":"","userId":""}},"outputId":"fb63aba3-eb1b-4924-90bf-b683822bd556","colab":{"base_uri":"https://localhost:8080/","height":333}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["--2023-10-24 15:38:20--  https://raw.githubusercontent.com/barrantesisrael/mbtmicrobiome2023/main/materials/setup_conda\n","Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.110.133, 185.199.108.133, ...\n","Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 2219 (2.2K) [text/plain]\n","Saving to: ‘setup_conda’\n","\n","\rsetup_conda           0%[                    ]       0  --.-KB/s               \rsetup_conda         100%[===================>]   2.17K  --.-KB/s    in 0s      \n","\n","2023-10-24 15:38:20 (31.7 MB/s) - ‘setup_conda’ saved [2219/2219]\n","\n"]},{"output_type":"display_data","data":{"text/plain":["\u001b[2;36m[15:38:27]\u001b[0m\u001b[2;36m \u001b[0m🐍 Downloading miniconda\u001b[33m...\u001b[0m                                                               \u001b]8;id=802947;file:///content/setup_conda\u001b\\\u001b[2msetup_conda\u001b[0m\u001b]8;;\u001b\\\u001b[2m:\u001b[0m\u001b]8;id=897160;file:///content/setup_conda#44\u001b\\\u001b[2m44\u001b[0m\u001b]8;;\u001b\\\n"],"text/html":["<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #7fbfbf; text-decoration-color: #7fbfbf\">[15:38:27] </span>🐍 Downloading miniconda<span style=\"color: #808000; text-decoration-color: #808000\">...</span>                                                               <a href=\"file:///content/setup_conda\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">setup_conda</span></a><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">:</span><a href=\"file:///content/setup_conda#44\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">44</span></a>\n","</pre>\n"]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["\u001b[2;36m[15:38:28]\u001b[0m\u001b[2;36m \u001b[0m\u001b[34m🐍 Done.\u001b[0m                                                                                  \u001b]8;id=881897;file:///content/setup_conda\u001b\\\u001b[2msetup_conda\u001b[0m\u001b]8;;\u001b\\\u001b[2m:\u001b[0m\u001b]8;id=256110;file:///content/setup_conda#50\u001b\\\u001b[2m50\u001b[0m\u001b]8;;\u001b\\\n"],"text/html":["<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #7fbfbf; text-decoration-color: #7fbfbf\">[15:38:28] </span><span style=\"color: #000080; text-decoration-color: #000080\">🐍 Done.</span>                                                                                  <a href=\"file:///content/setup_conda\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">setup_conda</span></a><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">:</span><a href=\"file:///content/setup_conda#50\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">50</span></a>\n","</pre>\n"]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["\u001b[2;36m          \u001b[0m\u001b[2;36m \u001b[0m🐍 Installing miniconda\u001b[33m...\u001b[0m                                                                \u001b]8;id=562960;file:///content/setup_conda\u001b\\\u001b[2msetup_conda\u001b[0m\u001b]8;;\u001b\\\u001b[2m:\u001b[0m\u001b]8;id=137183;file:///content/setup_conda#44\u001b\\\u001b[2m44\u001b[0m\u001b]8;;\u001b\\\n"],"text/html":["<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #7fbfbf; text-decoration-color: #7fbfbf\">           </span>🐍 Installing miniconda<span style=\"color: #808000; text-decoration-color: #808000\">...</span>                                                                <a href=\"file:///content/setup_conda\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">setup_conda</span></a><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">:</span><a href=\"file:///content/setup_conda#44\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">44</span></a>\n","</pre>\n"]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["\u001b[2;36m[15:38:41]\u001b[0m\u001b[2;36m \u001b[0m\u001b[34m🐍 Installed miniconda to `\u001b[0m\u001b[34m/usr/\u001b[0m\u001b[34mlocal\u001b[0m\u001b[34m`.\u001b[0m                                                   \u001b]8;id=935123;file:///content/setup_conda\u001b\\\u001b[2msetup_conda\u001b[0m\u001b]8;;\u001b\\\u001b[2m:\u001b[0m\u001b]8;id=27296;file:///content/setup_conda#50\u001b\\\u001b[2m50\u001b[0m\u001b]8;;\u001b\\\n"],"text/html":["<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #7fbfbf; text-decoration-color: #7fbfbf\">[15:38:41] </span><span style=\"color: #000080; text-decoration-color: #000080\">🐍 Installed miniconda to `/usr/local`.</span>                                                   <a href=\"file:///content/setup_conda\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">setup_conda</span></a><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">:</span><a href=\"file:///content/setup_conda#50\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">50</span></a>\n","</pre>\n"]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["\u001b[2;36m          \u001b[0m\u001b[2;36m \u001b[0m🧹 Cleaned up unneeded files.                                                             \u001b]8;id=286963;file:///content/setup_conda\u001b\\\u001b[2msetup_conda\u001b[0m\u001b]8;;\u001b\\\u001b[2m:\u001b[0m\u001b]8;id=812697;file:///content/setup_conda#39\u001b\\\u001b[2m39\u001b[0m\u001b]8;;\u001b\\\n"],"text/html":["<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #7fbfbf; text-decoration-color: #7fbfbf\">           </span>🧹 Cleaned up unneeded files.                                                             <a href=\"file:///content/setup_conda\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">setup_conda</span></a><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">:</span><a href=\"file:///content/setup_conda#39\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">39</span></a>\n","</pre>\n"]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["\u001b[2;36m          \u001b[0m\u001b[2;36m \u001b[0m\u001b[32mEverything is A-OK. You can start using conda now 👍\u001b[0m                                      \u001b]8;id=906928;file:///content/setup_conda\u001b\\\u001b[2msetup_conda\u001b[0m\u001b]8;;\u001b\\\u001b[2m:\u001b[0m\u001b]8;id=129088;file:///content/setup_conda#79\u001b\\\u001b[2m79\u001b[0m\u001b]8;;\u001b\\\n"],"text/html":["<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #7fbfbf; text-decoration-color: #7fbfbf\">           </span><span style=\"color: #008000; text-decoration-color: #008000\">Everything is A-OK. You can start using conda now 👍</span>                                      <a href=\"file:///content/setup_conda\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">setup_conda</span></a><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">:</span><a href=\"file:///content/setup_conda#79\" target=\"_blank\"><span style=\"color: #7f7f7f; text-decoration-color: #7f7f7f\">79</span></a>\n","</pre>\n"]},"metadata":{}}]},{"cell_type":"markdown","metadata":{"id":"IA376ZS9LZuw"},"source":["Afterwards we install all the programs we need today using `conda` (usually takes up to 5 minutes):"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"1WxfEpTlLboZ"},"outputs":[],"source":["! conda install -y -c bioconda fastqc pandaseq kraken2 >/dev/null 2>&1"]},{"cell_type":"markdown","source":["Let's verify our installs by running **_one_** the following commands:"],"metadata":{"id":"LfLQd1AASfte"}},{"cell_type":"code","source":["! fastqc -h"],"metadata":{"id":"rvBgB4AeVUrv"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! pandaseq -h"],"metadata":{"id":"FicfgnBgSlkc"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! kraken2"],"metadata":{"id":"sXkcb86MSnWQ","executionInfo":{"status":"ok","timestamp":1698162732897,"user_tz":-120,"elapsed":5,"user":{"displayName":"","userId":""}},"outputId":"bd759f99-d709-48f2-a4c0-3665f3cde925","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Need to specify input filenames!\n","Usage: kraken2 [options] <filename(s)>\n","\n","Options:\n","  --db NAME               Name for Kraken 2 DB\n","                          (default: none)\n","  --threads NUM           Number of threads (default: 1)\n","  --quick                 Quick operation (use first hit or hits)\n","  --unclassified-out FILENAME\n","                          Print unclassified sequences to filename\n","  --classified-out FILENAME\n","                          Print classified sequences to filename\n","  --output FILENAME       Print output to filename (default: stdout); \"-\" will\n","                          suppress normal output\n","  --confidence FLOAT      Confidence score threshold (default: 0.0); must be\n","                          in [0, 1].\n","  --minimum-base-quality NUM\n","                          Minimum base quality used in classification (def: 0,\n","                          only effective with FASTQ input).\n","  --report FILENAME       Print a report with aggregrate counts/clade to file\n","  --use-mpa-style         With --report, format report output like Kraken 1's\n","                          kraken-mpa-report\n","  --report-zero-counts    With --report, report counts for ALL taxa, even if\n","                          counts are zero\n","  --report-minimizer-data With --report, report minimizer and distinct minimizer\n","                          count information in addition to normal Kraken report\n","  --memory-mapping        Avoids loading database into RAM\n","  --paired                The filenames provided have paired-end reads\n","  --use-names             Print scientific names instead of just taxids\n","  --gzip-compressed       Input files are compressed with gzip\n","  --bzip2-compressed      Input files are compressed with bzip2\n","  --minimum-hit-groups NUM\n","                          Minimum number of hit groups (overlapping k-mers\n","                          sharing the same minimizer) needed to make a call\n","                          (default: 2)\n","  --help                  Print this message\n","  --version               Print version information\n","\n","If none of the *-compressed flags are specified, and the filename provided\n","is a regular file, automatic format detection is attempted.\n"]}]},{"cell_type":"markdown","source":["To conclude the setup, let's retrieve our course repository to a new folder called \"**course**\""],"metadata":{"id":"qxKnRhUYUVQA"}},{"cell_type":"code","execution_count":null,"metadata":{"id":"4uaEiRzaE4X9","executionInfo":{"status":"ok","timestamp":1698162740420,"user_tz":-120,"elapsed":2232,"user":{"displayName":"","userId":""}},"outputId":"7d6f4303-ee5f-47a8-b050-955a809ef39f","colab":{"base_uri":"https://localhost:8080/"}},"outputs":[{"output_type":"stream","name":"stdout","text":["Cloning into 'course'...\n","remote: Enumerating objects: 695, done.\u001b[K\n","remote: Counting objects: 100% (260/260), done.\u001b[K\n","remote: Compressing objects: 100% (156/156), done.\u001b[K\n","remote: Total 695 (delta 151), reused 186 (delta 103), pack-reused 435\u001b[K\n","Receiving objects: 100% (695/695), 5.38 MiB | 7.09 MiB/s, done.\n","Resolving deltas: 100% (345/345), done.\n"]}],"source":["! git clone https://github.com/barrantesisrael/mbt.microbiome.2021 course"]},{"cell_type":"markdown","source":["and retrieve the 16S Greengenes database by changing first to the materials folder, and running the `getgg.sh` script as following:"],"metadata":{"id":"_eKswEpDhrNi"}},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":338,"status":"ok","timestamp":1698162745043,"user":{"displayName":"","userId":""},"user_tz":-120},"id":"XC9ZCNCaIv5-","outputId":"42e1f3c9-3008-47c3-8dea-4b8b10306afe"},"outputs":[{"output_type":"stream","name":"stdout","text":["/content/course/data2023\n"]}],"source":["%cd course/data2023"]},{"cell_type":"code","source":["! bash getgg.sh"],"metadata":{"id":"KAyWve04hrt0"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"NmWoUIIuH0oe"},"source":["Notice that we used `%` instead of `!` to run out command line function. This makes the path change to our directory permanent. Check the content of the current directory/folder"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"01O1qXzTIAzK","executionInfo":{"status":"ok","timestamp":1698162780044,"user_tz":-120,"elapsed":380,"user":{"displayName":"","userId":""}},"outputId":"32aedd5c-9adc-4c9b-feab-bd3c58dc0ba0","colab":{"base_uri":"https://localhost:8080/"}},"outputs":[{"output_type":"stream","name":"stdout","text":["16S_Greengenes_k2db\t      Platz15_R2.head.fastq  Platz25_R1.head.fastq\n","bash_session_20231018_V01.sh  Platz16_R1.head.fastq  Platz25_R2.head.fastq\n","emptyfile\t\t      Platz16_R2.head.fastq  Platz2_R1.head.fastq\n","getgg.sh\t\t      Platz17_R1.head.fastq  Platz2_R2.head.fastq\n","KontrolleA_R1.head.fastq      Platz17_R2.head.fastq  Platz3_R1.head.fastq\n","KontrolleA_R2.head.fastq      Platz18_R1.head.fastq  Platz3_R2.head.fastq\n","KontrolleK_R1.head.fastq      Platz18_R2.head.fastq  Platz4_R1.head.fastq\n","KontrolleK_R2.head.fastq      Platz19_R1.head.fastq  Platz4_R2.head.fastq\n","mbtmicrobiome2023.biom\t      Platz19_R2.head.fastq  Platz5_R1.head.fastq\n","mbtmicrobiome2023.tsv\t      Platz1_R1.head.fastq   Platz5_R2.head.fastq\n","Platz10_R1.head.fastq\t      Platz1_R2.head.fastq   Platz6_R1.head.fastq\n","Platz10_R2.head.fastq\t      Platz20_R1.head.fastq  Platz6_R2.head.fastq\n","Platz11_R1.head.fastq\t      Platz20_R2.head.fastq  Platz7_R1.head.fastq\n","Platz11_R2.head.fastq\t      Platz21_R1.head.fastq  Platz7_R2.head.fastq\n","Platz12_R1.head.fastq\t      Platz21_R2.head.fastq  Platz8_R1.head.fastq\n","Platz12_R2.head.fastq\t      Platz22_R1.head.fastq  Platz8_R2.head.fastq\n","Platz13_R1.head.fastq\t      Platz22_R2.head.fastq  Platz9_R1.head.fastq\n","Platz13_R2.head.fastq\t      Platz23_R1.head.fastq  Platz9_R2.head.fastq\n","Platz14_R1.head.fastq\t      Platz23_R2.head.fastq  r_session_20231025_V01.R\n","Platz14_R2.head.fastq\t      Platz24_R1.head.fastq  sample-metadata-2023.tsv\n","Platz15_R1.head.fastq\t      Platz24_R2.head.fastq  sample-metadata.tsv\n"]}],"source":["! ls"]},{"cell_type":"markdown","metadata":{"id":"H6V82UcLIbXY"},"source":["Replace the name of the FASTQ file with your own identifier, and observe the first ten lines -what do you notice?"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"15iTtkwhIqXA","executionInfo":{"status":"ok","timestamp":1698162784078,"user_tz":-120,"elapsed":249,"user":{"displayName":"","userId":""}},"outputId":"bc1decd8-c4aa-44cc-8355-d62426a0556d","colab":{"base_uri":"https://localhost:8080/"}},"outputs":[{"output_type":"stream","name":"stdout","text":["@M02093:262:000000000-L53B2:1:1101:19849:1170 1:N:0:ACTCGCTA+CTCTCTAT\n","CCTACGGGCTGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAAGGAAGAAGTATTTCGGTATGTAAACTTCTATCAGCAGGGAAGATAGTGACGGTACCTGACTAAGAAGCCCCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGG\n","+\n","ACCCCFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGG>EGGGGFGGGGGGGGEGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGFEFGGFDDBGEFFEDEFGFF<?AF<BF<ADFFFFF>).1F????DFFBB>0:A04?BFFFF>3=:8)).))6<A)44<>F24<9B96>B\n","@M02093:262:000000000-L53B2:1:1101:19552:1182 1:N:0:ACTCGCTA+CTCTCTAT\n","CCTACAGGAGGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAGTGAAGAAGTATTTCGGTATGTAAAGCTCTATCAGCAGGGAAGAAAATGACGGTACCTGACTAAGAAGCCCTGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGAATTACTGGGTGTAAAGGGTGCGGAGGTGGTATGGCAAGTCAGAAGTGAAAACCCAGGGCTTAACTCTGGGACTGCTTTTGAAACTGTCAGACTGGAGTGCAGGAGAG\n","+\n","CCCCCG<FFFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGCGGGGGGGGFGGGGGDFGCGGGGGGGGGGGGCEGGGGGFFGEGGDEEGGGGGGGGGGGGGGGGGG@7FGGGGGFGGGG;;>D)-9D>FGF7=?FF<F>)7<?A>442>AB284;B:6D6(4:AF6<A923(2:?F>BD?AF?78AD<><AAF)3:1(..)65:(,(4\n","@M02093:262:000000000-L53B2:1:1101:21461:1244 1:N:0:ACTCGCTA+CTCTCTAT\n","CCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGACGAGAGTCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATACGGGAATAAAGTGAGGCACGTGTGCCTTTTTGTATGTACCGTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGGCGGACGCTTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGG\n"]}],"source":["! head Platz1_R1.head.fastq"]},{"cell_type":"markdown","metadata":{"id":"9xbxww1uI5oq"},"source":["\n","Count the total number of lines with the command below -how many READS are in this FASTQ? Hint: each read in FASTQ format consists of four lines\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"lDdFR3tcI7wE","executionInfo":{"status":"ok","timestamp":1698162788434,"user_tz":-120,"elapsed":395,"user":{"displayName":"","userId":""}},"outputId":"d7c078e9-af86-4ca8-a8e7-40a382504d22","colab":{"base_uri":"https://localhost:8080/"}},"outputs":[{"output_type":"stream","name":"stdout","text":["400 Platz1_R1.head.fastq\n"]}],"source":["! wc -l Platz1_R1.head.fastq"]},{"cell_type":"markdown","metadata":{"id":"iZFSR76oI9-R"},"source":["Find specific nucleotide combinations, e.g \"AATATT\""]},{"cell_type":"code","execution_count":null,"metadata":{"id":"_aczx-E7J5o4","executionInfo":{"status":"ok","timestamp":1698162791094,"user_tz":-120,"elapsed":363,"user":{"displayName":"","userId":""}},"outputId":"d19c8286-e71c-4b3d-e23a-cdfb653509dd","colab":{"base_uri":"https://localhost:8080/"}},"outputs":[{"output_type":"stream","name":"stdout","text":["CCTACGGGCTGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAAGGAAGAAGTATTTCGGTATGTAAACTTCTATCAGCAGGGAAGATAGTGACGGTACCTGACTAAGAAGCCCCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGG\n","CCTACAGGAGGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAGTGAAGAAGTATTTCGGTATGTAAAGCTCTATCAGCAGGGAAGAAAATGACGGTACCTGACTAAGAAGCCCTGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGAATTACTGGGTGTAAAGGGTGCGGAGGTGGTATGGCAAGTCAGAAGTGAAAACCCAGGGCTTAACTCTGGGACTGCTTTTGAAACTGTCAGACTGGAGTGCAGGAGAG\n","CCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGACGAGAGTCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATACGGGAATAAAGTGAGGCACGTGTGCCTTTTTGTATGTACCGTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGGCGGACGCTTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGG\n","CCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGCCGAGAGGCTGAACCAGCCAAGTCGCGTGAAGGAAGAAGGATCTATGGTTTGTAAACTTCTTTTATAGGGGAATAAAGTGGAGGACGTGTCCTTTTTTGTATGTACCCTATGAATAAGCATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGGGGTGATTTAAGTCAGCGGTGAAAGTTTGTGGCTCAACCATACAATTGCCGTTGAAACTGG\n","CCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGCCGAGAGGCTGAACCAGCCAAGTCGCGTGAAGGAAGAAGGATCTATGGTTTGTAAACTTCTTTTATAGGGGAATAAAGTGGAGGACGTGTCCTTTTTTGTATGTACCCTATGAATAAGCATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGTGGTGATTTAAGTCAGCGGTGAAAGTTTGTGGCTCAACCATAAAATTGCCGTTGAAACTGG\n","CCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGCCGAGAGGCTGAACCAGCCAAGTCGCGTGAAGGAAGAAGGATCTATGGTTTGTAAACTTCTTTTATAGGGGAATAAAGTGGAGGACGTGTCCTTTTTTGTATGTACCCTATGAATAAGCATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGTGGTGATTTAAGTCAGCGGTGAAAGTTTGTGGCTCAACCATAAAATTGCCGTTGAAACTGG\n","CCTACGGGGCGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAGCGAAGAAGTATTTCGGTATGTAAACTTCTATCAGCAGGGAAGACAGTGACGGTACCTGACTAAGAAGCCCCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTGGCAAGTCTGATGTGAAAGGCATGGGCTCAACCTGTGGACTGCATTGGAAACTGTCATACTTGAGTGCCGGAGGG\n","CCTACGGGTGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATAAAGGAATAAAGTCGGGTATGCATACCCGTTTGCATGTACTTTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGATGGATGTTTAAGTCAGTTGTGAAAGTTTGCGGCCCAACCGTAAAATTGCAGTTGATACTGG\n","CCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGCCGAGAGGCTGAACCAGCCAAGTCGCGTGAGGGATGAAGGTTCTATGGATCGTAAACCTCTTTTATAAGGGAATAAAGTGTGGGACGTGTCCTATTTTGTATGTACCTTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGCCTTTTAAGTCAGCGGTGAAAGTCTGTGGCTCAACCCTAGAATTGCAGTGGAAACTGG\n","CCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGCCGAGAGGCTGAACCAGCCAAGTCGCGTGAGGGATGAAGGTTCTATGGATCGTAAACCTCTTTTATAAGGGAATAAAGTGTGGGACGTGTCCTATTTTGTATGTACCTTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGCCTTTTAAGTCAGCGGTGAAAGTCTGTGGCTCAACCATAGAATTGCCGTTGAAAATGG\n"]}],"source":["! grep \"AATATT\" Platz1_R1.head.fastq | head"]},{"cell_type":"markdown","metadata":{"id":"uKFg6yTPJ7ov"},"source":["Count how many times the hexanucleotide \"AATATT\" appears in your Illumina data"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"2Hw2lyjlKFv4","executionInfo":{"status":"ok","timestamp":1698162794729,"user_tz":-120,"elapsed":437,"user":{"displayName":"","userId":""}},"outputId":"d57ecdae-93fc-44a3-984e-83ba9140f85f","colab":{"base_uri":"https://localhost:8080/"}},"outputs":[{"output_type":"stream","name":"stdout","text":["87\n"]}],"source":["! grep -c \"AATATT\" Platz1_R1.head.fastq"]},{"cell_type":"markdown","source":["### 1.2 Quality control with the [FASTQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) tool\n","\n","Replace the `Platz` number with your own in the following command before running"],"metadata":{"id":"oQsTp6tbUIHo"}},{"cell_type":"code","source":["! fastqc --quiet Platz1_R1.head.fastq"],"metadata":{"id":"U7dzclbAjD1R","executionInfo":{"status":"ok","timestamp":1698162802879,"user_tz":-120,"elapsed":5264,"user":{"displayName":"","userId":""}},"outputId":"a6a77dae-14df-4204-ffb0-72322cac3afe","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["null\n"]}]},{"cell_type":"markdown","source":["To display the output, download the HTML output from the `Files` view on the side menu to your local computer, and open it in a web browser.\n","\n","\n","\n","### 1.3 Amplicon assembly with [pandaseq](https://github.com/neufeld/pandaseq)\n","\n","Again replace the `Platz` number with your own in the following command before running the following command:"],"metadata":{"id":"oaC-894Vj42k"}},{"cell_type":"code","source":["! pandaseq -f Platz10_R1.head.fastq -r Platz10_R2.head.fastq -w Platz10.fa -g log.txt"],"metadata":{"id":"Kfy_aE5O1SEt"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Observe the first ten lines of the FASTA output:"],"metadata":{"id":"cQzFjgiz1S1I"}},{"cell_type":"code","source":["! head Platz10.fa"],"metadata":{"id":"XL0G7EH11fha","executionInfo":{"status":"ok","timestamp":1698162817769,"user_tz":-120,"elapsed":235,"user":{"displayName":"","userId":""}},"outputId":"d6fec76a-a6e9-46ec-c1a5-f50eb18ef079","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":[">M02093:262:000000000-L53B2:1:1101:10955:1184:ACTCGCTA+ACTGCATA\n","CCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAAGGAAGAAGTATCTCGGTATGTAAACTTCTATCAGCAGGGAAGAAAATGACGGTACCTGACTAAGAAGCCCCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTTGCAAGTCTGATGTGAAAGACGGGGGCTCAACCCCTGGACTGCATTGGAAACTGTGATACTCGAGTGCCGGAGAGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCGTGTAGTC\n",">M02093:262:000000000-L53B2:1:1101:16471:1188:ACTCGCTA+ACTGCATA\n","CCTAGGGAGGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCAACGCCGCGTGAGTGATGACGGCCTTCGGGTTGTAAAGCTCTGTCTTCAGGGACGATAATGACGGTACCTGAGGAGGAAGCCACGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTATGTCGCAAGCGTTATCCGGATTTATTGGGCGTAAAGCGCGTCTAGGCGGTCTGGTAAGTCTGATGTGGAAATGCGGGGCTCAACTCCGTATTGCGTTGGAAACTGCCAGACTAGAGTACTGGAGAGGTGGGCGGAACTACAAGTGTAGAGGTGAAAGTCGTAGATATTTGTAGGAATGCCGATAGAGAAGTCAGCTCACTGGACAGATACTGACGCTGAAGCGCGAAAGCATGGGAGCAAACAGGATTAGATACCCTCGTAGTC\n",">M02093:262:000000000-L53B2:1:1101:18772:1190:ACTCGCTA+ACTGCATA\n","CCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCAACGCCGCGTGAGTGATGACGGCCTTCGGGTTGTAAAGCTCTGTCTTCAGGGACGATAATGACGGTACCTGAGGAGGAAGCCACGGCTAACTACGTGTCAGCAGCCGCGGTAATACGTAGGTGGCGAGCGTTGTCCGGATTTACTGGGCGTAAAGGGAGCGTAGGCGGACTTTTAAGTGAGATGTGAAATACCCGGGCTCAACTTGGGTGCTGCATTTCAAACTGGAAGTCTAGAGTGCAGGAGAGGAGAATGGAATTCCTAGTGTAGCGGTGAAATGCGTAGAGATTAGGAAGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTCGTAGTC\n",">M02093:262:000000000-L53B2:1:1101:15482:1194:ACTCGCTA+ACTGCATA\n","CCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGGCGTTAGCCTGAACCAGCCAAGTAGCGTGAAGGATGAAGGCTCTATGGGTCGTAAACTTCTTTTATATAAGAATAAAGTGCAGTATGTATACTGTTTTGTATGTATTATATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGGTGGACTGGTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGTCAGTCTTGAGTACAGTAGAGGTGGGCGGAATTCGTGGTGTAGCGGTGAAATGCTTAGATATCACGAAGAACTCCGATTGCGAAGGCAGCTCACTGGACTGCAACTGACACTGATGCTCGAAAGTGTGGGTATCAAACAGGATTAGATACCCTTGTAGTC\n",">M02093:262:000000000-L53B2:1:1101:10965:1202:ACTCGCTA+ACTGCATA\n","CCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAAGGAAGAAGTATCTCGGTATGTAAACTTCTATCAGCAGGGAAGAAAATGACGGTACCTGACTAATAAGCCCCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTGTTGCAAGTCTGATGTGAAAGACGGGGGCTCAACCCCTGGACTGCATTGGAAACTGTGATACTCGAGTGCCGGAGAGGGAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGACGGTAACTGACGTTGAGGCTCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCGTGTAGTC\n"]}]},{"cell_type":"markdown","source":["Q: _How many sequences are in this FASTA output?_\n","\n","Hint: count the total number of HEADER line symbols (\">\") with the command:\n"],"metadata":{"id":"jIWjZ4Wb1jtY"}},{"cell_type":"code","source":["! grep -c \">\" Platz10.fa"],"metadata":{"id":"5bfsgs3_1vzo","executionInfo":{"status":"ok","timestamp":1698162822028,"user_tz":-120,"elapsed":271,"user":{"displayName":"","userId":""}},"outputId":"186ecb38-347e-443b-879c-6b69815d8f97","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["97\n"]}]},{"cell_type":"markdown","source":["Q: _What is the rate of FASTA sequences vs FASTQ reads? And what does this tell about our sequencing and assembly quality and efficiency?_\n","\n","\n","\n","### 1.3 OTU Assignment with [kraken2](https://ccb.jhu.edu/software/kraken2/) against the 16S [Greengenes](https://greengenes.secondgenome.com/) database\n","\n","Run your samples against Greengenes with your own data; the example here is the with the `Platz10` run, so replace the FASTQ filenames with your own FASTQ names as before.\n","\n"],"metadata":{"id":"fJ-WgFDn2K88"}},{"cell_type":"code","source":["! kraken2 --db 16S_Greengenes_k2db --use-names --output output.txt --report report.txt --paired Platz10_R1.head.fastq Platz10_R2.head.fastq"],"metadata":{"id":"Ab8T-jeR27NC","executionInfo":{"status":"ok","timestamp":1698162825659,"user_tz":-120,"elapsed":239,"user":{"displayName":"","userId":""}},"outputId":"b01d97c2-7c2e-4884-ec10-4e1aa0adcde6","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Loading database information... done.\n","\rProcessed 100 sequences (60094 bp) ...\r100 sequences (0.06 Mbp) processed in 0.012s (511.4 Kseq/m, 307.31 Mbp/m).\n","  100 sequences classified (100.00%)\n","  0 sequences unclassified (0.00%)\n"]}]},{"cell_type":"markdown","source":["Inspect your individual results (file: `report.txt`) with the following command:"],"metadata":{"id":"jVSwTv0j3Wc1"}},{"cell_type":"code","source":["! cat report.txt"],"metadata":{"id":"bSoy3N7v3W6A","executionInfo":{"status":"ok","timestamp":1698162871493,"user_tz":-120,"elapsed":541,"user":{"displayName":"","userId":""}},"outputId":"20592d50-bd34-4b04-c76b-349a61f01f30","colab":{"base_uri":"https://localhost:8080/"}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["100.00\t100\t0\tR\t1\troot\n","100.00\t100\t0\tD\t3\t  Bacteria\n"," 48.00\t48\t0\tP\t34\t    Firmicutes\n"," 43.00\t43\t0\tC\t226\t      Clostridia\n"," 43.00\t43\t6\tO\t527\t        Clostridiales\n"," 25.00\t25\t11\tF\t1017\t          Lachnospiraceae\n","  5.00\t5\t5\tG\t1800\t            Coprococcus\n","  4.00\t4\t4\tG\t1797\t            Blautia\n","  4.00\t4\t4\tG\t1801\t            Dorea\n","  1.00\t1\t0\tG\t1810\t            [Ruminococcus]\n","  1.00\t1\t1\tS\t2783\t              [Ruminococcus] gnavus\n","  7.00\t7\t2\tF\t1020\t          Ruminococcaceae\n","  4.00\t4\t4\tG\t1830\t            Oscillospira\n","  1.00\t1\t1\tG\t1831\t            Ruminococcus\n","  3.00\t3\t1\tF\t1010\t          Clostridiaceae\n","  2.00\t2\t0\tG\t1777\t            Clostridium\n","  2.00\t2\t2\tS\t2770\t              Clostridium perfringens\n","  2.00\t2\t0\tF\t1025\t          Veillonellaceae\n","  2.00\t2\t2\tG\t1840\t            Dialister\n","  5.00\t5\t0\tC\t225\t      Bacilli\n","  5.00\t5\t0\tO\t524\t        Lactobacillales\n","  4.00\t4\t0\tF\t1003\t          Enterococcaceae\n","  4.00\t4\t4\tG\t1756\t            Enterococcus\n","  1.00\t1\t0\tF\t1006\t          Streptococcaceae\n","  1.00\t1\t1\tG\t1765\t            Streptococcus\n"," 37.00\t37\t0\tP\t17\t    Bacteroidetes\n"," 37.00\t37\t0\tC\t169\t      Bacteroidia\n"," 37.00\t37\t1\tO\t423\t        Bacteroidales\n"," 29.00\t29\t0\tF\t884\t          Bacteroidaceae\n"," 29.00\t29\t16\tG\t1497\t            Bacteroides\n","  7.00\t7\t7\tS\t2596\t              Bacteroides fragilis\n","  3.00\t3\t3\tS\t2597\t              Bacteroides ovatus\n","  3.00\t3\t3\tS\t2599\t              Bacteroides uniformis\n","  6.00\t6\t0\tF\t888\t          Porphyromonadaceae\n","  5.00\t5\t1\tG\t1501\t            Parabacteroides\n","  4.00\t4\t4\tS\t2601\t              Parabacteroides distasonis\n","  1.00\t1\t1\tG\t1500\t            Paludibacter\n","  1.00\t1\t1\tF\t892\t          S24-7\n","  9.00\t9\t0\tP\t67\t    Proteobacteria\n","  6.00\t6\t0\tC\t294\t      Betaproteobacteria\n","  6.00\t6\t0\tO\t606\t        Burkholderiales\n","  6.00\t6\t0\tF\t1101\t          Alcaligenaceae\n","  6.00\t6\t6\tG\t2064\t            Sutterella\n","  3.00\t3\t1\tC\t297\t      Gammaproteobacteria\n","  1.00\t1\t0\tO\t658\t        Chromatiales\n","  1.00\t1\t1\tF\t1168\t          Chromatiaceae\n","  1.00\t1\t0\tO\t659\t        Enterobacteriales\n","  1.00\t1\t1\tF\t1171\t          Enterobacteriaceae\n","  4.00\t4\t0\tP\t35\t    Fusobacteria\n","  4.00\t4\t0\tC\t229\t      Fusobacteriia\n","  4.00\t4\t0\tO\t536\t        Fusobacteriales\n","  4.00\t4\t0\tF\t1044\t          Fusobacteriaceae\n","  4.00\t4\t4\tG\t1912\t            Fusobacterium\n","  2.00\t2\t0\tP\t11\t    Actinobacteria\n","  2.00\t2\t0\tC\t146\t      Coriobacteriia\n","  2.00\t2\t0\tO\t407\t        Coriobacteriales\n","  2.00\t2\t0\tF\t867\t          Coriobacteriaceae\n","  2.00\t2\t0\tG\t1478\t            Collinsella\n","  1.00\t1\t1\tS\t2587\t              Collinsella aerofaciens\n","  1.00\t1\t1\tS\t2588\t              Collinsella stercoris\n"]}]},{"cell_type":"markdown","source":["Question: _What are the most predominant genera in your personal Illumina runs?_\n","\n","The output report from Kraken2 consists of the following fields:\n","\n","- Percentage of fragments covered by the clade rooted at this taxon\n","- Number of fragments covered by the clade rooted at this taxon\n","- Number of fragments assigned directly to this taxon\n","- A rank code, indicating (U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies.\n","- NCBI taxonomic ID number\n","- Indented scientific name\n","\n","See the program [documentation](https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown#sample-report-output-format) for details.\n","\n"],"metadata":{"id":"sE5k1aiM3HDH"}}],"metadata":{"colab":{"provenance":[{"file_id":"https://github.com/barrantesisrael/mbtmicrobiome2023/blob/main/materials/bashsession_mbt2023_v01.ipynb","timestamp":1698162884336}],"authorship_tag":"ABX9TyNlJzU6FtlfGCBANUlq7TdV"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}