{"cells":[{"cell_type":"markdown","metadata":{"id":"XUmYvAk6yri8"},"source":["# UMR MBT Microbiome Praktikum\n","\n","---\n","\n","## Session 1\n","\n","\n","##### 1.1 Setup and introduction to the command line for Bioinformatics\n","\n","First, let's run the following cell, to install the `conda` package manager\n"]},{"cell_type":"code","source":["! wget https://raw.githubusercontent.com/barrantesisrael/mbtmicrobiome2023/main/materials/setup_conda\n","%run setup_conda"],"metadata":{"id":"bgY2B4gZOnfS"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"IA376ZS9LZuw"},"source":["Afterwards we install all the programs we need today using `conda` (usually takes up to 5 minutes):"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"1WxfEpTlLboZ"},"outputs":[],"source":["! conda install -y -c bioconda fastqc pandaseq kraken2 >/dev/null 2>&1"]},{"cell_type":"markdown","source":["Let's verify our installs by running **_one_** the following commands:"],"metadata":{"id":"LfLQd1AASfte"}},{"cell_type":"code","source":["! fastqc -h"],"metadata":{"id":"rvBgB4AeVUrv"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! pandaseq -h"],"metadata":{"id":"FicfgnBgSlkc"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! kraken2"],"metadata":{"id":"sXkcb86MSnWQ"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["To conclude the setup, let's retrieve our course repository to a new folder called \"**course**\""],"metadata":{"id":"qxKnRhUYUVQA"}},{"cell_type":"code","execution_count":null,"metadata":{"id":"4uaEiRzaE4X9"},"outputs":[],"source":["! git clone https://github.com/barrantesisrael/mbt.microbiome.2021 course"]},{"cell_type":"markdown","source":["and retrieve the 16S Greengenes database by changing first to the materials folder, and running the `getgg.sh` script as following:"],"metadata":{"id":"_eKswEpDhrNi"}},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":281,"status":"ok","timestamp":1697819096947,"user":{"displayName":"Israel Barrantes","userId":"01503052389382241223"},"user_tz":-120},"id":"XC9ZCNCaIv5-","outputId":"fb396d65-33f5-4834-a0a9-e6328ddda028"},"outputs":[{"output_type":"stream","name":"stdout","text":["/content/course/data2023\n"]}],"source":["%cd course/data2023"]},{"cell_type":"code","source":["! bash getgg.sh"],"metadata":{"id":"KAyWve04hrt0"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"NmWoUIIuH0oe"},"source":["Notice that we used `%` instead of `!` to run out command line function. This makes the path change to our directory permanent. Check the content of the current directory/folder"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"01O1qXzTIAzK"},"outputs":[],"source":["! ls"]},{"cell_type":"markdown","metadata":{"id":"H6V82UcLIbXY"},"source":["Replace the name of the FASTQ file with your own identifier, and observe the first ten lines -what do you notice?"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"15iTtkwhIqXA"},"outputs":[],"source":["! head Platz1_R1.head.fastq"]},{"cell_type":"markdown","metadata":{"id":"9xbxww1uI5oq"},"source":["\n","Count the total number of lines with the command below -how many READS are in this FASTQ? Hint: each read in FASTQ format consists of four lines\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"lDdFR3tcI7wE"},"outputs":[],"source":["! wc -l Platz1_R1.head.fastq"]},{"cell_type":"markdown","metadata":{"id":"iZFSR76oI9-R"},"source":["Find specific nucleotide combinations, e.g \"AATATT\""]},{"cell_type":"code","execution_count":null,"metadata":{"id":"_aczx-E7J5o4"},"outputs":[],"source":["! grep \"AATATT\" Platz1_R1.head.fastq | head"]},{"cell_type":"markdown","metadata":{"id":"uKFg6yTPJ7ov"},"source":["Count how many times the hexanucleotide \"AATATT\" appears in your Illumina data"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"2Hw2lyjlKFv4"},"outputs":[],"source":["! grep -c \"AATATT\" Platz1_R1.head.fastq"]},{"cell_type":"markdown","source":["### 1.2 Quality control with the [FASTQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) tool\n","\n","Replace the `Platz` number with your own in the following command before running"],"metadata":{"id":"oQsTp6tbUIHo"}},{"cell_type":"code","source":["! fastqc --quiet Platz1_R1.head.fastq"],"metadata":{"id":"U7dzclbAjD1R"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["To display the output, download the HTML output from the `Files` view on the side menu to your local computer, and open it in a web browser.\n","\n","\n","\n","### 1.3 Amplicon assembly with [pandaseq](https://github.com/neufeld/pandaseq)\n","\n","Again replace the `Platz` number with your own in the following command before running the following command:"],"metadata":{"id":"oaC-894Vj42k"}},{"cell_type":"code","source":["! pandaseq -f Platz10_R1.head.fastq -r Platz10_R2.head.fastq -w Platz10.fa -g log.txt"],"metadata":{"id":"Kfy_aE5O1SEt"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Observe the first ten lines of the FASTA output:"],"metadata":{"id":"cQzFjgiz1S1I"}},{"cell_type":"code","source":["! head Platz10.fa"],"metadata":{"id":"XL0G7EH11fha"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Q: _How many sequences are in this FASTA output?_\n","\n","Hint: count the total number of HEADER line symbols (\">\") with the command:\n"],"metadata":{"id":"jIWjZ4Wb1jtY"}},{"cell_type":"code","source":["! grep -c \">\" Platz10.fa"],"metadata":{"id":"5bfsgs3_1vzo"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Q: _What is the rate of FASTA sequences vs FASTQ reads? And what does this tell about our sequencing and assembly quality and efficiency?_\n","\n","\n","\n","### 1.3 OTU Assignment with [kraken2](https://ccb.jhu.edu/software/kraken2/) against the 16S [Greengenes](https://greengenes.secondgenome.com/) database\n","\n","Run your samples against Greengenes with your own data; the example here is the with the `Platz10` run, so replace the FASTQ filenames with your own FASTQ names as before.\n","\n"],"metadata":{"id":"fJ-WgFDn2K88"}},{"cell_type":"code","source":["! kraken2 --db 16S_Greengenes_k2db --use-names --output output.txt --report report.txt --paired Platz10_R1.head.fastq Platz10_R2.head.fastq"],"metadata":{"id":"Ab8T-jeR27NC"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Inspect your individual results (file: `report.txt`) with the following command:"],"metadata":{"id":"jVSwTv0j3Wc1"}},{"cell_type":"code","source":["! cat report.txt"],"metadata":{"id":"bSoy3N7v3W6A"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Question: _What are the most predominant genera in your personal Illumina runs?_\n","\n","The output report from Kraken2 consists of the following fields:\n","\n","- Percentage of fragments covered by the clade rooted at this taxon\n","- Number of fragments covered by the clade rooted at this taxon\n","- Number of fragments assigned directly to this taxon\n","- A rank code, indicating (U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies.\n","- NCBI taxonomic ID number\n","- Indented scientific name\n","\n","See the program [documentation](https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown#sample-report-output-format) for details.\n","\n","\n","---\n","\n","## Contact\n","\n","Dr. rer. nat. Israel Barrantes <br>\n","Junior Research Group Translational Bioinformatics (head)<br>\n","Institute for Biostatistics and Informatics in Medicine and Ageing Research, Office 3017<br>\n","Rostock University Medical Center<br>\n","Ernst-Heydemann-Str. 8<br>\n","18057 Rostock, Germany<br>\n","\n","Email: israel.barrantes[bei]uni-rostock.de\n","\n","---\n","Last update 2023/10/12"],"metadata":{"id":"sE5k1aiM3HDH"}}],"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyOCgex3qib3gzFPeMe3jWSt"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}