{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[]},"kernelspec":{"name":"ir","display_name":"R"},"language_info":{"name":"R"}},"cells":[{"cell_type":"markdown","source":["# UMR MBT Microbiome Praktikum\n","\n","---\n","\n","## Session 2\n","\n","##### 2.1 Loading libraries and microbiome data #####\n","\n","First, we start our R environment by loading the needed packages. For this, we need to install first Bioconductor and then the `phyloseq` library."],"metadata":{"id":"ubz70rxcKreY"}},{"cell_type":"code","source":["install.packages(\"BiocManager\")\n","BiocManager::install(\"phyloseq\")"],"metadata":{"id":"uyt7YxdXL82z"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Now we load our required libraries:"],"metadata":{"id":"GS5LpHU7Qs6i"}},{"cell_type":"code","source":["# load ggplot2 library (graphics)\n","library(ggplot2)\n","\n","# loading phyloseq library (microbiome analysis)\n","library(phyloseq)"],"metadata":{"id":"HjPF0W2_KygV"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Then we load the needed input files: the OTU table (BIOM) and the metadata (TSV)"],"metadata":{"id":"Iw-4c81bK8So"}},{"cell_type":"code","source":["# OTU data\n","InputBiomFile <- \"https://raw.githubusercontent.com/barrantesisrael/mbt.microbiome.2021/main/data2023/mbtmicrobiome2023.biom\"\n","\n","# Samples' metadata\n","InputMapFile <- \"https://raw.githubusercontent.com/barrantesisrael/mbt.microbiome.2021/main/data2023/sample-metadata-2023.tsv\"\n","\n","# prepare phyloseq object by loading both files\n","BiomData <- import_biom(InputBiomFile, parseFunction = parse_taxonomy_greengenes)\n","SampleData <- import_qiime_sample_data(InputMapFile)"],"metadata":{"id":"mUV0jWOvLFBx"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Afterwards, we merge these files into a single `phyloseq` object:"],"metadata":{"id":"GxxEaX_YLSzC"}},{"cell_type":"code","source":["# create phyloseq object by merging OTU and sample data\n","ExperimentPhyloseqObject <- merge_phyloseq(BiomData, SampleData)"],"metadata":{"id":"Am7vZF2sLZaw"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["We call our object to see its properties:"],"metadata":{"id":"iZ7i0-J_Ll_c"}},{"cell_type":"code","source":["# checking the features of our original microbiome data\n","ExperimentPhyloseqObject"],"metadata":{"id":"9rRJj_xNLs2y"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Now we create another phyloseq object, that includes only those taxa with more than 100 amplicons in all samples, and with `Gender` in the metadata"],"metadata":{"id":"gWEDNlM4SE_g"}},{"cell_type":"code","source":["# create a temporary phyloseq object for working\n","psTemp <- ExperimentPhyloseqObject\n","\n","# subset samples\n","# Prune OTUs with low abundances from all samples\n","psTemp <- prune_taxa(taxa_sums(psTemp) > 100, psTemp)\n","\n","# Prune samples with no metadata\n","psTemp <-  subset_samples(psTemp, Gender != \"NA\")\n","\n","# checking the features of our microbiome data\n","psTemp"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":90},"id":"A1S08FTDSIye","outputId":"bc09745c-8489-4cc6-8e47-5d331cd58dd7"},"execution_count":null,"outputs":[{"output_type":"display_data","data":{"text/plain":["phyloseq-class experiment-level object\n","otu_table()   OTU Table:         [ 557 taxa and 24 samples ]\n","sample_data() Sample Data:       [ 24 samples by 6 sample variables ]\n","tax_table()   Taxonomy Table:    [ 557 taxa by 7 taxonomic ranks ]"]},"metadata":{}}]},{"cell_type":"markdown","source":["_Q: What are the differences between the `ExperimentPhyloseqObject` and `psTemp` objects?_\n","\n","##### 2.2 Sample ordination #####"],"metadata":{"id":"r7gUWt21SYcl"}},{"cell_type":"code","source":["# Calculate distance and ordination\n","iDist <- distance(psTemp, method=\"bray\")\n","iMDS  <- ordinate(psTemp, distance=iDist)\n","\n","# plot sample ordination, e.g. by Gender\n","plot_ordination(psTemp, iMDS, color=\"Gender\")"],"metadata":{"id":"Cz1iiWNbS2ZC"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Q: Are there any clear separations between the gender groups? Why/Why not?\n","\n","Repeat the ordination using other metadata, e.g. \"Cat\":"],"metadata":{"id":"iQXBO3YFS5-O"}},{"cell_type":"code","source":["plot_ordination(psTemp, iMDS, color=\"Cat\") +\n","  geom_text(aes(label=SampleID), vjust = -1)"],"metadata":{"id":"LB9m9JP9TIAr"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##### 2.3 Microbial communities #####\n","\n","First let's observe the absolute abundance on each sample at the Phylum level."],"metadata":{"id":"wAsNM8PZThs_"}},{"cell_type":"code","source":["### Abundances per sample\n","plot_bar(psTemp, \"SampleID\", fill=\"Phylum\")"],"metadata":{"id":"vd_oGes8ToXk"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["As these are not directly comparable, we will perform a rarefaction, as a means to normalize the data, and then plot again"],"metadata":{"id":"cTAS7aqKUCSd"}},{"cell_type":"code","source":["# Rarefaction to an even depth\n","ps.rarefied <- rarefy_even_depth(psTemp)\n","\n","# Remove lines\n","ps.rarefied.glom <- tax_glom(ps.rarefied, \"Phylum\")\n","\n","# Plot abundances\n","plot_bar(ps.rarefied.glom, \"SampleID\", fill=\"Phylum\")"],"metadata":{"id":"RA9ZP5VpUL4y"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["Q: _What are the differences between the abundance plots before and after rarefaction?_\n","\n","We can also plot the relative abundances according to metadata, e.g. by `Gender`:"],"metadata":{"id":"f2z12499UTWX"}},{"cell_type":"code","source":["### Abundances per category, e.g. \"Gender\"\n","# Merge samples by the selected category\n","mergedGP <- merge_samples(psTemp, \"Gender\")\n","\n","# Rarefaction to an even depth\n","ps.rarefied <- rarefy_even_depth(mergedGP)\n","\n","# Remove lines\n","ps.rarefied.glom <- tax_glom(ps.rarefied, \"Phylum\")\n","\n","# Plot abundances for the example category \"Gender\"\n","plot_bar(ps.rarefied.glom, fill=\"Phylum\")"],"metadata":{"id":"r8GIdGZTUt7-"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["##### 2.4 Diversity #####\n","\n","First let's observe the Shannon diversity on the individual samples:"],"metadata":{"id":"VwD_gB65U2a9"}},{"cell_type":"code","source":["plot_richness(psTemp, x = \"SampleID\", measures = c(\"Shannon\"))"],"metadata":{"id":"PCSwduUwU90Z"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":[" Repeat the same analyses at the Gender level:"],"metadata":{"id":"ilJAKBJ0U-mZ"}},{"cell_type":"code","source":["plot_richness(psTemp, x = \"Gender\", color = \"Gender\", measures = c(\"Shannon\"))"],"metadata":{"id":"63xsdhy5VEc6"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["A more appropriate view of our plot can be achieved by adding more features, e.g.:"],"metadata":{"id":"YH3qiP7uVHXn"}},{"cell_type":"code","source":["# Assign our plot to a variable\n","Our_Richness_plot <- plot_richness(psTemp, x = \"Gender\", color = \"Gender\", measures = c(\"Shannon\"))\n","\n","# Improving our plot e.g. by adding proper labels\n","Our_Richness_plot + geom_boxplot(data = Our_Richness_plot$data, aes(x = Gender, y = value, color = Gender), alpha = 0.1) + # boxplot\n","  labs(title = \"Richness (Shannon alpha diversity)\", subtitle = \"MBT Class 2023\") + # title and subtitle\n","  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) # x-axis labels: 0 degree rotation, 0.5 horizontal position"],"metadata":{"id":"HjjTaZ8mVW3T"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["---\n","\n","## Contact\n","\n","Dr. rer. nat. Israel Barrantes <br>\n","Junior Research Group Translational Bioinformatics (head)<br>\n","Institute for Biostatistics and Informatics in Medicine and Ageing Research, Office 3017<br>\n","Rostock University Medical Center<br>\n","Ernst-Heydemann-Str. 8<br>\n","18057 Rostock, Germany<br>\n","\n","Email: israel.barrantes[bei]uni-rostock.de\n","\n","---\n","Last update 2023/10/12"],"metadata":{"id":"AToQLIm9RWit"}}]}